Source/KakapoServer.swift

```diff
@@ -91,11 +103,15 @@ public final class KakapoServer: NSURLProtocol {
     
     /// Start loading the matched requested, the route handler will be called and the returned object will be serialized.
     override public func startLoading() {
-        KakapoServer.routers.filter { $0.canInitWithRequest(request) }.first!.startLoading(self)
+        if let routerIndex = KakapoServer.routers.indexOf({ $0.canInitWithRequest(request) }) {
+            KakapoServer.routers[routerIndex].startLoading(self)
+        }
     }
     
-    /// Not implemented yet, does nothing ATM. https://github.com/devlucky/Kakapo/issues/88
+    /// Stops the loading of the matched request.
     override public func stopLoading() {
-        /* TODO: implement stopLoading for delayed requests https://github.com/devlucky/Kakapo/issues/88 */
+        if let routerIndex = KakapoServer.routers.indexOf({ $0.canInitWithRequest(request) }) {
+            KakapoServer.routers[routerIndex].stopLoading(self)
+        }
```

Comment:
I've added a test case in ea5a34962367d80ef8aa62877934c5dbda991d8c for the case when the URLRequest gets cancelled directly.

However, `Router` is handling the trigger of the `NSURLProtocolClient#URLProtocolDidFinishLoading...` delegate call. Apple doc requires that no further delegate calls are performed when the request has been stopped.
Soo, I added some test cases for this behaviour as well, which basically ensures that the `URLProtocolDidFinishLoading` does not get called by the `Router` when the request has been marked as `stopped` by the OS.